pkg/transforms/mqttsecret.go

```diff
@@ -336,3 +336,10 @@ func (sender *MQTTSecretSender) ConnectToBroker(lc logger.LoggingClient, sp boot
 		sender.preConnected = true
 	}
 }
+
+// Set MQTT client options before calling ConnectToBroker.
+func (sender *MQTTSecretSender) SetClientOpts(opts *MQTT.ClientOptions) {
+	onConHandler := opts.OnConnect
+	sender.opts.SetOnConnectHandler(onConHandler)
+
+}
```

Comment:
I have 2 issues here. 

1.  This overrides the OnConnect already being set here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/pkg/transforms/mqttsecret.go#L100 which is just for trace logging. Documentation should cover this and suggest add the same trace logging to the custom onConnect handler.
2. Since this is only setting the OnConnect handler the function should be more explicit
```suggestion
// SetOnConnect sets the OnConnect Handler before client is connected so client can be captured.
func (sender *MnQTTSecretSender) SetOnConnect(onConnect mqtt.OnConnectHandler ) {
	sender.opts.SetOnConnectHandler(onConnect)
}
```

Reply:
> I have 2 issues here. 
> 
> 
> 
> 1.  This overrides the OnConnect already being set here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/pkg/transforms/mqttsecret.go#L100 which is just for trace logging. Documentation should cover this and suggest add the same trace logging to the custom onConnect handler.
> 
> 2. Since this is only setting the OnConnect handler the function should be more explicit
> 
> ```suggestion
> 
> // SetOnConnect sets the OnConnect Handler before client is connected so client can be captured.
> 
> func (sender *MnQTTSecretSender) SetOnConnect(onConnect mqtt.OnConnectHandler ) {
> 
> 	sender.opts.SetOnConnectHandler(onConnect)
> 
> }
> 
> ```

@lenny-intel  Yes,you're right. It's my fault, I only focused on my purpose. 
I just want to set the `OnConnectHandler` opt, and the business logic will be explain in my application service. I'll check the code again.